### PR TITLE
Fix type stubs in locks.pyi

### PR DESCRIPTION
**Pull request by smessmer**

### CODE REVIEW

File: stdlib/3.4/asyncio/locks.pyi [hunk 1]
```diff
@@ -49,11 +51,11 @@ class Condition(_ContextManagerMixin):
     def notify_all(self) -> None: ...
 
 class Semaphore(_ContextManagerMixin):
-    def __init__(self, value: int = ..., *, loop: Optional[AbstractEventLoop] = ...) -> None: ...
+    def __init__(self, value: int = 1, *, loop: AbstractEventLoop = None) -> None: ...
```
Comment (matthiaskramm) [inline stdlib/3.4/asyncio/locks.pyi]: This looks odd. Why did you change `...` to "1" and "None"? We generally prefer the former, in typeshed.

File: stdlib/3.4/asyncio/locks.pyi [hunk 2]
```diff
@@ -17,8 +19,8 @@ class _ContextManagerMixin(Future[_ContextManager]):
     # Apparently this exists to *prohibit* use as a context manager.
     def __enter__(self) -> object: ...
     def __exit__(self, *args: Any) -> None: ...
-    def __aenter__(self): ...
-    def __aexit__(self, exc_type, exc, tb): ...
+    def __aenter__(self) -> Awaitable[object]: ...
+    def __aexit__(self, exc_type: Optional[type], exc: Optional[BaseException], tb: Optional[TracebackType]) -> Awaitable[None]: ...
```
Comment (JelleZijlstra) [inline stdlib/3.4/asyncio/locks.pyi]: The type should be `Optional[Type[BaseException]]`.

File: stdlib/3.4/asyncio/locks.pyi [hunk 3]
```diff
@@ -17,8 +19,8 @@ class _ContextManagerMixin(Future[_ContextManager]):
     # Apparently this exists to *prohibit* use as a context manager.
     def __enter__(self) -> object: ...
     def __exit__(self, *args: Any) -> None: ...
-    def __aenter__(self): ...
-    def __aexit__(self, exc_type, exc, tb): ...
+    def __aenter__(self) -> Awaitable[object]: ...
```
Comment (JelleZijlstra) [inline stdlib/3.4/asyncio/locks.pyi]: I might need to think more about whether and how it makes a difference, but we've usually annotated coroutine methods with `@types.coroutine` instead of using Awaitable as the return type.

Also, we should avoid putting the `object` there. I haven't looked at the code yet, but maybe we can use a self type?

File: stdlib/3.4/asyncio/locks.pyi [hunk 4]
```diff
@@ -49,11 +51,11 @@ class Condition(_ContextManagerMixin):
...
-    def __init__(self, value: int = ..., *, loop: Optional[AbstractEventLoop] = ...) -> None: ...
+    def __init__(self, value: int = 1, *, loop: AbstractEventLoop = None) -> None: ...
     def locked(self) -> bool: ...
     @coroutine
     def acquire(self) -> Generator[Any, None, bool]: ...
     def release(self) -> None: ...
 
 class BoundedSemaphore(Semaphore):
-    def __init__(self, value: int = ..., *, loop: Optional[AbstractEventLoop] = ...) -> None: ...
+    def __init__(self, value: int = 1, *, loop: AbstractEventLoop = None) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3.4/asyncio/locks.pyi]: Matthias's comment applies here too. We'll likely remove the paragraph in PEP 484 soon that says that setting the default to None implies making the argument Optional.

**Comment by JelleZijlstra** [comment]
@matthiaskramm can you take another look and merge if you think it's ready? I fixed the requested changes.
